spring-test/src/main/java/org/springframework/mock/web/MockHttpServletRequest.java

```diff
@@ -912,6 +928,9 @@ public long getDateHeader(String name) {
 		else if (value instanceof Number) {
 			return ((Number) value).longValue();
 		}
+		else if (value instanceof String) {
+			return parseDateHeader(name, (String) value);
+		}
 		else if (value != null) {
 			throw new IllegalArgumentException(
 					"Value for header '" + name + "' is neither a Date nor a Number: " + value);
```

Comment:
Need to update exception message as follows:

`"Value for header '" + name + "' is not a Date, Number, or String: " + value`